File: Daily/sbgo/Leetcode278-第一个错误的版本.go
package sbgo

func firstBadVersion(n int) int {
	l, r := 1, n
	for l < r {
		mid := (l + r) >> 1
		if !isBadVersion(mid) {
			l = mid + 1
		} else {
			r = mid
		}
	}
	return l
}

File: Leetcode/PY/Leetcode278-第一个错误的版本.py
class Solution:
    def firstBadVersion(self, n):
        """
        :type n: int
        :rtype: int
        """
        l,r = 1,n
        while l < r:
            mid = l + r >> 1
            if not isBadVersion(mid):
                l = mid + 1
            else:
                r = mid
        return l